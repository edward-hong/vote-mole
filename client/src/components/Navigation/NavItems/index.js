import React from 'react'
import { connect } from 'react-redux'
import { NavLink, NavItem } from 'reactstrap'
import styled from 'styled-components'
import { isNil } from 'ramda'

import { AUTH_LOGOUT_PATH } from '../../../constants'

const NavLinkItem = styled(NavItem)`
	text-align: center;
`

// Render nothing if user info hasn't been fetched from backend
// Then conditionally render NavItems depending on whether user is logged in
const NavItems = ({ toggleLogin, toggleAddPoll, isLoggedIn }) =>
	isNil(isLoggedIn) ? null : isLoggedIn ? (
		<>
			<NavLink href="#" onClick={toggleAddPoll}>
				<NavLinkItem>Add Poll</NavLinkItem>
			</NavLink>
			<NavLink href={AUTH_LOGOUT_PATH}>
				<NavLinkItem>Logout</NavLinkItem>
			</NavLink>
		</>
	) : (
		<NavLink href="#" onClick={toggleLogin}>
			<NavLinkItem>Login</NavLinkItem>
		</NavLink>
	)

const mapStateToProps = ({ auth }) => ({ isLoggedIn: auth })

export default connect(mapStateToProps)(NavItems)
