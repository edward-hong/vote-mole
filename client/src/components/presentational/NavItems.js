import React from 'react'
import { connect } from 'react-redux'
import { NavLink, NavItem } from 'reactstrap'
import styled from 'styled-components'
import { isNil } from 'ramda'

const NavLinkItem = styled(NavItem)`
	text-align: center;
`

const NavItems = ({ toggleLogin, toggleAddPoll, isLoggedIn }) =>
	isNil(isLoggedIn) ? null : isLoggedIn ? (
		<>
			<NavLink href="#" onClick={toggleAddPoll}>
				<NavLinkItem>Add Poll</NavLinkItem>
			</NavLink>
			<NavLink href="/auth/logout">
				<NavLinkItem>Logout</NavLinkItem>
			</NavLink>
		</>
	) : (
		<NavLink href="#" onClick={toggleLogin}>
			<NavLinkItem>Login</NavLinkItem>
		</NavLink>
	)

const mapStateToProps = ({ auth }) => ({ isLoggedIn: auth })

export default connect(mapStateToProps)(NavItems)
