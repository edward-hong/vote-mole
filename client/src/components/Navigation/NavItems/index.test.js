import React from 'react'

import NavItems from '.'
import renderWithRedux from '../../../utils/test/renderWithRedux'
import renderWithRouterAndRedux from '../../../utils/test/renderWithRouterAndRedux'

describe('NavItems', () => {
	it('returns null when `auth` state is null', () => {
		const { container } = renderWithRedux(
			<NavItems toggleAddPoll={() => {}} toggleLogin={() => {}} />,
			{ initialState: { auth: null } },
		)

		expect(container.firstChild).toBeNull()
	})

	it('renders correctly when `auth` state is false', () => {
		const { container } = renderWithRouterAndRedux(
			<NavItems toggleAddPoll={() => {}} toggleLogin={() => {}} />,
			{ initialState: { auth: false } },
		)

		expect(container.firstChild).toMatchSnapshot()
	})

	it('renders correctly when `auth` state is true', () => {
		const auth = {
			_id: '5cd19682498a98c6962f34c1',
			profileID: '111881712224957728911',
		}
		const { container } = renderWithRouterAndRedux(
			<NavItems toggleAddPoll={() => {}} toggleLogin={() => {}} />,
			{ initialState: { auth } },
		)

		expect(container.firstChild).toMatchSnapshot()
	})
})
