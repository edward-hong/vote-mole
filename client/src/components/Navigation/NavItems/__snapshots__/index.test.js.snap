// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItems renders correctly when \`auth\` state is false 1`] = `
.c0 {
  text-align: center;
}

<a
  class="nav-link"
  href="#"
>
  <li
    class="c0 nav-item"
  >
    Login
  </li>
</a>
`;

exports[`NavItems renders correctly when \`auth\` state is true 1`] = `
.c0 {
  text-align: center;
}

<a
  class="nav-link"
  href="/my_polls"
>
  <li
    class="c0 nav-item"
  >
    My Polls
  </li>
</a>
`;
